go/sentry/client/client.go

```diff
@@ -0,0 +1,137 @@
+// Package client implements a client for Oasis sentry nodes.
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/balancer/roundrobin"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/resolver"
+
+	"github.com/oasislabs/oasis-core/go/common/grpc/resolver/manual"
+	"github.com/oasislabs/oasis-core/go/common/identity"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	"github.com/oasislabs/oasis-core/go/common/node"
+	"github.com/oasislabs/oasis-core/go/grpc/sentry"
+	"github.com/oasislabs/oasis-core/go/sentry/api"
+)
+
+var (
+	_ api.Backend = (*Client)(nil)
+)
+
+var (
+	// ErrSentryNotAvailable is the error returned when the sentry node is not
+	// available.
+	ErrSentryNotAvailable = errors.New("sentry/client: sentry node not available")
+)
+
+type Client struct {
+	ctx context.Context
```

Comment:
It would be better if `GetConsensusAddresses` took a `context.Context` argument instead of storing a context here.